### ISSUE TITLE
Symlinks with stack group config doesn't work

### ISSUE THREAD

**Issue by et304383** [cloudreach/sceptre#634]
```
tree:

.
├── config
│   ├── base
│   │   ├── config.yaml
│   │   └── dynamo.yaml
│   ├── config.yaml
│   └── dev -> base
└── templates
    └── dynamo.j2

[dynamo_tables]$ sceptre2 validate dev
Traceback (most recent call last):
  File "/home/et304383/Programs/python_virtualenvs/sceptre2/bin/sceptre", line 10, in <module>
    sys.exit(cli())
  File "/home/et304383/Programs/python_virtualenvs/sceptre2/lib/python3.6/site-packages/click/core.py", line 722, in __call__
    return self.main(*args, **kwargs)
  File "/home/et304383/Programs/python_virtualenvs/sceptre2/lib/python3.6/site-packages/click/core.py", line 697, in main
    rv = self.invoke(ctx)
  File "/home/et304383/Programs/python_virtualenvs/sceptre2/lib/python3.6/site-packages/click/core.py", line 1066, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/home/et304383/Programs/python_virtualenvs/sceptre2/lib/python3.6/site-packages/click/core.py", line 895, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/et304383/Programs/python_virtualenvs/sceptre2/lib/python3.6/site-packages/click/core.py", line 535, in invoke
    return callback(*args, **kwargs)
  File "/home/et304383/Programs/python_virtualenvs/sceptre2/lib/python3.6/site-packages/click/decorators.py", line 17, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "/home/et304383/Programs/python_virtualenvs/sceptre2/lib/python3.6/site-packages/sceptre/cli/helpers.py", line 37, in decorated
    return func(*args, **kwargs)
  File "/home/et304383/Programs/python_virtualenvs/sceptre2/lib/python3.6/site-packages/sceptre/cli/template.py", line 35, in validate_command
    responses = plan.validate()
  File "/home/et304383/Programs/python_virtualenvs/sceptre2/lib/python3.6/site-packages/sceptre/plan/plan.py", line 318, in validate
    return self._execute(*args)
  File "/home/et304383/Programs/python_virtualenvs/sceptre2/lib/python3.6/site-packages/sceptre/plan/plan.py", line 36, in _execute
    executor = SceptrePlanExecutor(self.command, self.launch_order)
  File "/home/et304383/Programs/python_virtualenvs/sceptre2/lib/python3.6/site-packages/sceptre/plan/executor.py", line 35, in __init__
    self.num_threads = len(max(launch_order, key=len))
ValueError: max() arg is an empty sequence

[dynamo_tables]$ sceptre2 validate base
'project_code' is undefined
```

This worked fine in sceptre 1.  We do this because most test environments are identical except for the environment name (stack group name).

**Comment by ngfgrant** [comment]
An immediate fix for this would be to reset your symbolic links using their absolute path. I will see if we can fix for relative paths later although Sceptre path handling now bases itself off of absolute paths.

**Comment by et304383** [comment]
Huh?  You're saying, just don't use symlinks is the solution to symlinks being broken?

What changed in V2 to break this?

**Comment by ngfgrant** [comment]
no I mean you can use symbolic links but you will need to potentially re-add them but instead of using relative paths using the abs path. Example:

```tree
.
├── config
│   ├── config.yaml
│   ├── dev
│   │   ├── config.yaml
│   │   └── x.yaml
│   └── prod
│       └── zz.yaml -> /Users/niallgrant/Codebase/sceptre/sceptre-examples/test-new/testing-new/config/dev/x.yaml
└── templates
    ├── vpc.json
    └── vpc.yaml
```

**Comment by vaneek** [comment]
That sounds like a bit of a kludge and will break any portability and automation. I use sceptre in codebuild and have no desire determine and rewrite links everytime a checkout and build is performed.
I may revert back to using Makefiles.

**Comment by ngfgrant** [comment]
Will definitely investigate this further - it is likely to be some python path shenanigans.

**Comment by davidtaddei** [comment]
@et304383 could you possibly provide your OS version and exact python version down to the patch level please? This may help others reproduce the issue. I have attempted your setup in MacOS and it seems to work. I also tried symbolically linking individual yaml files which also worked. I'll check my exact python version and get an inventory of everything in my virtual environment. Thanks!

**Comment by et304383** [comment]
Ubuntu 18.04

I'm using a virtualenv:

```
[et304383: sceptre2]$ . bin/activate
(sceptre2) [et304383: sceptre2]$ python -V
Python 3.6.7
(sceptre2) [et304383: sceptre2]$ sceptre --version
Sceptre, version 2.0.2
(sceptre2) [et304383: sceptre2]$ 
```

**Comment by ngfgrant** [comment]
resolved by #635
